### ISSUE TITLE
Consider a smarter bootstrap script

### ISSUE THREAD

**Issue by jamiemccarthy** [voxmedia/docker_base_images#18]
After conversation with @thenonsequitur, I'm withdrawing #16 and filing this issue instead.

Basically the idea is that there are things that should be automated before a developer can run `docker-compose` for a project. Things that can, maybe should, run outside containers. A big one is setting up an ssh exec obviously. But there's other things too. Anthem has a bootstrap script. Let's think about whether that's something to make generic for other projects.

Slack conversation was:

> Jamie McCarthy [2 hours ago] 
> the case I would make is
> 
> 
> Jamie McCarthy [2 hours ago] 
> Sanity checks are almost never a bad idea. This one is going to be transparent (it runs quickly even on large repos) and if it catches an issue it's 99% likely to be a real issue. So maybe it doesn't matter much that people besides you are unlikely to trip it. Most rake tests never trip either. Tests are a good thing for their own sake.
> 
> 
> Ben Lee [2 hours ago] 
> I disagree that tests are good a thing for their own sake, they add code, which adds complexity. There is always a cost. But the specific sanity check is less of an issue in my mind than the sliippery slope of putting these checks in the wrong place.
> 
> 
> Ben Lee 
> [2 hours ago] 
> I have a meeting, will get back to this after.
> 
> Jamie McCarthy [2 hours ago] 
> ok
> 
> Jamie McCarthy [2 hours ago] 
> Re the slippery slope, like the PR writeup says, IMHO we should be taking even more out of the written instructions and putting them into scripts.
> Whether the script's name is correct is a separate issue, maybe it should reference any and all automated pre-flighting we do before docker build, rather than the single task of starting a single script which hopefully will eventually be obviated anyway.
> Changing the script's name is probably out of scope though.
> 
> Ben Lee [2 hours ago] 
> don't disagree that we should be automating steps. disagree that it makes sense to just add them somewhere.
> 
> Ben Lee 
> [2 hours ago] 
> i really don't like hijacking this script for this purpose
> 
> Ben Lee [2 hours ago] 
> i would rather ticket this issue for further thinking and designing than add this code here
> 
> Ben Lee 
> [2 hours ago] 
> (my meeting was cancelled btw)
> 
> Ben Lee [2 hours ago] 
> Especially considering this scirpt is only meant to be run once -- we'd want sanity checks like this to maybe be more thorough and frequent
> 
> Ben Lee 
> [2 hours ago] 
> TL;DR: I agree that sanity checks are a good thing and we should automate as many steps as possible.
> 
> I disagree that this is important enough to implement in a script that was designed for an entirely separate purpose rather than putting off until we can design a better place.
> 
> Ben Lee [2 hours ago] 
> I have been toying with the idea of applying an anthem-esque bootstrap script to other repos. Anthem's bootstrap was designed to run inside the containers but it would be easy to refactor to run outside the containers...and a script like that would be a much better place for sanity checks of this sort imo
> 
> Ben Lee 
> [2 hours ago] 
> it would also be the appropriate place to do docker-sync checks for sbn (edited)
> 
> 
> Jamie McCarthy [2 hours ago] 
> Sounds good. Let's table my PR then, and come back to this idea later

**Comment by thenonsequitur** [comment]
The bootstrap conversation is now part of the sandboxes track, and is covered by the dev plan.
